packages/editor-ui/src/components/ExpressionEdit.vue

```diff
@@ -124,6 +126,9 @@ export default defineComponent({
 	},
 	computed: {
 		...mapStores(useNDVStore, useWorkflowsStore),
+		isReadOnlyRoute() {
+			return this.$route?.meta?.readonly === true;
```

Comment:
This `readonly` state should always be passed down as a prop.. Otherwise, it would lead to bugs where this component is used on editable pages.
Just tested this component on the credentials page for example, and it was readonly.. So that's a bug, which seems to exist on master.

Might be worth fixing as part of this refactor or a quick follow up (your call). It was always too complicated to refactor because of the dependency hell that's NDV.
<img width="829" alt="Screenshot 2024-01-04 at 14 22 23" src="https://github.com/n8n-io/n8n/assets/4711238/714ccdb6-aab0-46e6-8965-280acfc3d4f5">

Reply:
I see the issue now. As a fix I've tried to:
1. Remove `isReadOnlyRoute` from `ExpressionEdit` and `NodeCredentials`
2. Add `isReadOnly` prop to `ExpressionEdit`

We're already passing `:is-read-only="isReadOnly"` in `ParameterInput` to `ExpressionEdit` and it seems to be working as expected from the tests I've run. WDYT?
https://github.com/n8n-io/n8n/pull/8220/commits/670999c84c7f19ee48960b9398c67e6195395fc7